fearless/Modules/Staking/StakingMain/View/AlertsView.swift

```diff
@@ -0,0 +1,228 @@
+        let label = UILabel()
+        label.font = .p2Paragraph
+        label.textColor = R.color.colorWhite()?.withAlphaComponent(0.64)
+        return label
+    }()
+
+    private let alertsStackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.axis = .vertical
+        return stackView
+    }()
+
+    var locale = Locale.current {
+        didSet {
+            if locale != oldValue {
+                applyLocalization()
+            }
+        }
+    }
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        applyLocalization()
+        setupLayout()
+    }
+
+    @available(*, unavailable)
+    required init?(coder _: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func applyLocalization() {
+        titleLabel.text = R.string.localizable.stakingAlertsTitle(preferredLanguages: locale.rLanguages)
+        // TODO: review mockup and delete noAlertsLabel?
+        noAlertsLabel.text = "Everything is fine now. Alerts will appear here."
+    }
+
+    private func setupLayout() {
+        addSubview(backgroundView)
+        backgroundView.snp.makeConstraints { $0.edges.equalToSuperview() }
+
+        addSubview(titleLabel)
+        titleLabel.snp.makeConstraints { make in
+            make.top.leading.equalToSuperview().inset(UIConstants.horizontalInset)
+        }
+
+        let separatorView = UIView.createSeparator(color: R.color.colorWhite()?.withAlphaComponent(0.24))
+        addSubview(separatorView)
+        separatorView.snp.makeConstraints { make in
+            make.top.equalTo(titleLabel.snp.bottom).offset(15)
+            make.leading.trailing.equalToSuperview().inset(UIConstants.horizontalInset)
+            make.height.equalTo(UIConstants.separatorHeight)
+        }
+
+        addSubview(noAlertsLabel)
+        noAlertsLabel.snp.makeConstraints { make in
+            make.top.equalTo(separatorView.snp.bottom).offset(UIConstants.horizontalInset)
+            make.leading.trailing.bottom.equalToSuperview().inset(UIConstants.horizontalInset)
+        }
+
+        addSubview(alertsStackView)
+        alertsStackView.snp.makeConstraints { make in
+            make.top.equalTo(separatorView.snp.bottom)
+            make.leading.trailing.bottom.equalToSuperview()
+        }
+    }
+
+    func bind(alerts: [StakingAlert]) {
+        if alerts.isEmpty {
+            noAlertsLabel.isHidden = false
+            alertsStackView.isHidden = true
+        } else {
+            noAlertsLabel.isHidden = true
+            alertsStackView.isHidden = false
+
+            let itemViews = alerts.map { alert -> UIView in
+                let itemView = AlertItemView(stakingAlert: alert, locale: locale)
+                itemView.addTarget(self, action: #selector(handleSelectItem), for: .touchUpInside)
+                return itemView
+            }
+
+            let separators = (0 ..< itemViews.count).map { _ -> UIView in
```

Comment:
I think we could have BorderedContainerView (appearance of the separator is controlled via borderType) inside AlertItemView